test/components/course-listing.spec.coffee

```diff
@@ -39,6 +40,21 @@ describe 'Course Listing Component', ->
       # no refresh button when load succeeds
       expect(state.div.querySelector(".refresh-button")).to.be.null
 
+  it 'renders the listing without archived courses', ->
+    courseList = _.flatten([MASTER_COURSES_LIST, STUDENT_ARCHIVED_COURSE])
+    CourseListingActions.loaded(courseList)
+    renderListing().then (state) ->
+      renderDataset = _.pluck(state.div.querySelectorAll('.tutor-booksplash-course-item'), 'dataset')
+      for course, i in MASTER_COURSES_LIST
+        expect(MASTER_COURSES_LIST[i].name).to.contain(renderDataset[i].title)
+      # no refresh button when load succeeds
+      expect(state.div.querySelector(".refresh-button")).to.be.null
+
+  it 'renders empty courses if course list only contains arhived course', ->
+    CourseListingActions.loaded([STUDENT_ARCHIVED_COURSE])
+    renderListing().then (state) ->
+      expect(state.div.querySelector('.-course-list-empty')).not.to.be.null
+
   it 'renders course appropriate help', ->
```

Comment:
Nice specs!  It'll be great to have tests of the archived course case, since we've had so many bugs around that